src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -294,3 +297,45 @@ CHIP_ERROR OperationalStateServer::Read(const ConcreteReadAttributePath & aPath,
     }
     return CHIP_NO_ERROR;
 }
+
+void OperationalStateServer::OnOperationalErrorDetect(const Structs::ErrorStateStruct::Type & aError)
+{
+    ChipLogDetail(Zcl, "OperationalStateServer: OnOperationalErrorDetect");
+    MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, OperationalState::Attributes::OperationalState::Id);
+
+    EventNumber eventNumber;
```

Comment:
Reimplementing all this stuff is not ideal, but I guess... I'm worried that this will be less maintainable than if we actually had a struct that looks like an event struct for the right cluster and then logged it via the normal machinery.  Is there a reason that was not done?